specs/upgrades.md

```diff
@@ -0,0 +1,387 @@
+2. We must convert the subnet ID to an ETH address. To do this, visit the [Beryx address converter](https://beryx.zondax.ch/address_converter) and input the `t410` address in the Filecoin address input, in this example this would be `t410fkp4r67rks3ok4bvbn2rjojhkbig2rwvp4nmor5q`. Click `Convert to ETH` and it will compute the Ethereum address as `0x53f91f7e2a96dcae06a16ea29724ea0a0da8daaf`.
+3. You must set the `RPC_URL` and `PRIVATE_KEY` environmental variables to point to your network provider and the private key of the address you want to use for the deployment, respectively
+
+Now you should be able to upgrade the subnet actor by running the following command (change NETWORK if you have deployed your subnet on another network)
+
+```bash
+> cd contracts/
+> make upgrade-sa-diamond SUBNET_ACTOR_ADDRESS=0x53f91f7e2a96dcae06a16ea29724ea0a0da8daaf NETWORK=calibrationnet
+```
+
+A successful output looks like:
+
+```bash
+Successfully generated 84 typings!
+Compiled 14 Solidity files successfully (evm target: paris).
+
+Facet Bytecode Not Found:
+---------------------------------
+Facet Name: SubnetActorGetterFacet
+Libraries:
+  - SubnetIDHelper: 0x1CaA8B5a18d0401f2089aC82063AaecA08142744
+Address: 0xcb4EE21d081e0cBdd3D82D94114bEAa497f11C4a
+
+Diamond Facet Upgrade:
+-----------------------------------
+Diamond Address: 0x53f91f7e2a96dcae06a16ea29724ea0a0da8daaf
+Replacement Facet Name: SubnetActorGetterFacet
+
+Deployment Status:
+-------------------------
+New replacement facet (SubnetActorGetterFacet) deployed.
+
+Facet Bytecode Not Found:
+---------------------------------
+Facet Name: SubnetActorCheckpointingFacet
+Libraries:
+
+Address: 0xd5C97F52030DFE7595b6cD92868EcE5dC175C224
+
+Diamond Facet Upgrade:
+-----------------------------------
+Diamond Address: 0x53f91f7e2a96dcae06a16ea29724ea0a0da8daaf
+Replacement Facet Name: SubnetActorCheckpointingFacet
+
+Deployment Status:
+-------------------------
+New replacement facet (SubnetActorCheckpointingFacet) deployed.
+```
+
+If you run the `make upgrade-sa-diamond` command again, you should see no output which would also confirm that the upgrade has succeeded.
+
+# Fendermint upgrades
+
+Fendermint incorporates a builtin `UpgradeScheduler`, enabling the execution of hardcoded `Upgrade` migrations at predetermined block heights to advance the on-chain state. Fendermint also supports functionality to halt at predetermined `halt_height` in order to switch binary versions.
+
+## Upgrade scheduler
+
+At the start of processing each block, Fendermint checks the `UpgradeScheduler` whether it contains an `Upgrade` for that block height. Only a single `Upgrade` can be scheduled for each block height. If an upgrade is found to be scheduled, it will execute the `migration` function that is associated with that `Upgrade`. If the upgrade migration returns an error it will crash the node.
```

Comment:
Can you add the link to code where "Fendermint checks the `UpgradeScheduler`" happens?